core/services/workflows/engine.go

```diff
@@ -1191,23 +1191,23 @@ func (e *Engine) Name() string {
 }
 
 type Config struct {
-	Workflow              sdk.WorkflowSpec
-	WorkflowID            string
-	WorkflowOwner         string
-	WorkflowName          string // Full human-readable workflow name. Intended for metrics and logging.
-	WorkflowNameTransform string // The Workflow Name in an on-chain format, which has requirements of being hex encoded and max 10 bytes
-	Lggr                  logger.Logger
-	Registry              core.CapabilitiesRegistry
-	MaxWorkerLimit        int
-	QueueSize             int
-	NewWorkerTimeout      time.Duration
-	MaxExecutionDuration  time.Duration
-	Store                 store.Store
-	Config                []byte
-	Binary                []byte
-	SecretsFetcher        secretsFetcher
-	HeartbeatCadence      time.Duration
-	StepTimeout           time.Duration
+	Workflow             sdk.WorkflowSpec
+	WorkflowID           string
+	WorkflowOwner        string
+	WorkflowName         string // Full human-readable workflow name. Intended for metrics and logging.
+	WorkflowHexName      string // The Workflow Name in an on-chain format, which has requirements of being 10 bytes long, hex encoded (i.e. 20 hex characters long). If not provided, we'll derive it from the workflowName.
```

Comment:
Did you consider using another type for this field? Something like `type WorkflowHexName [10]byte` with a `String() string` method would prevent mixing up the `string`s. Or you could even use an opaque type to include them both, e.g. `type WorkflowName struct { name string }` with `Workflow.String()` and `Workflow.Hex()` methods to force a choice :shrug:

Reply:
Or perhaps something in between, like a `type WorkflowName string` with `WorkflowName.Hex()` :thinking: